tests/e2e/browse-bills.spec.ts

```diff
@@ -0,0 +1,637 @@
+    const searchTerm = getRandomWord()
+
+    await performSearch(page, searchTerm)
+
+    const labelsTextContent = await getCategoryLabelsTextContent(page)
+
+    labelsTextContent.forEach((text, index) => {
+      console.log(`Label ${index + 1}: ${text}`)
+    })
+
+    const containsSearchTerm = labelsTextContent.some(textContent =>
+      textContent.toLowerCase().includes(searchTerm.toLowerCase())
+    )
+    expect(containsSearchTerm).toBe(true)
+  })
+
+  test("check the first bill on random page", async ({ page }) => {
+    // Perform a search and check the first bill on random pages multiple times
+    const searchTerm = getRandomWord()
+
+    await performSearch(page, searchTerm)
+
+    const initialResultCount = await page.textContent(
+      ".ResultCount__ResultContainer-sc-3931e200-0"
+    )
+
+    await waitForResultsToChange(page, initialResultCount!)
+
+    let times = 2
+    do {
+      await checkFirstBill(page, searchTerm)
+      await clickNextPageRandomTimes(page)
+      times--
+    } while (times > 0)
+  })
+
+  test("no results found", async ({ page }) => {
+    // Test to ensure the application handles no results found cases
+    const searchTerm = "nonexistentsearchterm12345"
+
+    await performSearch(page, searchTerm)
+
+    const noResultsMessage = await page.textContent(
+      ".NoResults__Container-sc-bf043801-0 .text-center"
+    )
+    expect(noResultsMessage).toContain("Your search has yielded zero results!")
+  })
+})
+
+// Define an interface for sorting test configurations
+interface SortingTest {
+  option: string
+  attribute: string
+  order: "asc" | "desc"
+  type:
+    | "relevance"
+    | "testimonyCount"
+    | "cosponsorCount"
+    | "nextHearingDate"
+    | "recentTestimony"
+}
+
+// Array of sorting test configurations
+const sortingTests: SortingTest[] = [
+  // {
+  //   option: "Sort by Relevance",
+  //   attribute: "data-relevance-score",
+  //   order: "desc",
+  //   type: "relevance"
+  // },
+  {
+    option: "Sort by Testimony Count",
+    attribute: "div.testimonyCount",
+    order: "desc",
+    type: "testimonyCount"
+  },
+  {
+    option: "Sort by Cosponsor Count",
+    attribute: "span.blurb",
+    order: "desc",
+    type: "cosponsorCount"
+  },
+  {
+    option: "Sort by Next Hearing Date",
+    attribute: "div.card-footer",
+    order: "asc",
+    type: "nextHearingDate"
+  },
+  {
+    option: "Sort by Most Recent Testimony",
+    attribute: "span.blurb.me-2",
+    order: "asc",
+    type: "recentTestimony"
+  }
+]
+
+/**
+ * Extracts attribute values based on the type of sorting.
+ * @param item - The DOM element from which to extract the value.
+ * @param attribute - The attribute or selector used to locate the value.
+ * @param type - The type of value to extract ('relevance', 'testimonyCount', 'cosponsorCount', 'nextHearingDate', or 'recentTestimony').
+ * @returns The extracted value as a number.
+ */
+const getAttributeValue = async (
+  item: ElementHandle<Element>,
```

Comment:
It looks like Playwright wants us to move away from `ElementHandle`s to using locators (https://playwright.dev/docs/api/class-elementhandle#element-handle-query-selector-all) - could we update this logic to handle that change?

Reply:
I wasn't aware that Playwright is moving away from ElementHandles in favor of using locators. It's good to know about this update!

I agree that we should switch our logic to use locators as recommended. I'll start updating the relevant parts of our codebase to handle this change.